### PR TITLE
Fixed #380 - hideExpression is now called on field's model change

### LINKED ISSUE #380: hideExpression not called when model property specified
I am not sure if this is correct, but I think hideExpression is not called on model update when the model property in field config is specified (different from the form model).

if I use expressionProperties: hide, it works ok.

### PR DESCRIPTION
**Pull request by redhead**

### CODE REVIEW

**Comment by redhead** [comment]
Please review the code carefully, I am not sure this should be the way to handle this issue.

File: src/directives/formly-form.test.js [hunk 1]
```diff
@@ -358,6 +358,25 @@ describe('formly-form', () => {
...
+      scope.model = {};
+      scope.fieldModel = {};
+
+      scope.fields = [
+        {template: input, key: 'foo', model: scope.fieldModel},
+        {template: input, key: 'bar', model: scope.fieldModel, hideExpression: () => !!scope.fieldModel.foo}
+      ];
+    });
+
+    it('should be updated when field model changes', () => {
```
Comment (kentcdodds) [inline src/directives/formly-form.test.js]: Fantastic! Could you add another test that checks the opposite case? Where the field's `hide` property is initialized to `true` and then is swapped to `false` via the `hideExpression`? I think that would be a valuable test case.

Comment (redhead) [inline src/directives/formly-form.test.js]: Yeah, sure.

File: src/directives/formly-form.js [hunk 2]
```diff
@@ -188,6 +188,9 @@ function formlyForm(formlyUsability, formlyWarn, $parse, formlyConfig, $interpol
             field);
         }
       }
+      if (field.model) {
+        $scope.$watch(() => field.model, onModelOrFormStateChange, true);
```
Comment (redhead) [inline src/directives/formly-form.js]: @kentcdodds I just realised, this line might trigger multiple watch handlers if multiple fields use the same model object. I don't know if that is wrong in particular though.
Maybe we should create a set of model objects and create always only a single watcher for each?

Comment (kentcdodds) [inline src/directives/formly-form.js]: Ah, very good. Sounds like a reasonable optimization to me. You'd probably need to do a separate `angular.forEach` above in that case. If you decide to do that, please write a test case to cover that :-)

Comment (redhead) [inline src/directives/formly-form.js]: Will do.

**Comment by kentcdodds** [comment]
Thanks for working on this. While you're making updates and such, could you make sure that the PR ends up with only a single commit? Let me know if you need help getting that to work.

**Pull request closed without merging**